core/templates/dev/head/pages/exploration_editor/ExplorationSaveAndPublishButtons.js

```diff
@@ -0,0 +1,69 @@
+oppia.controller('ExplorationSaveAndPublishButtons', [
+  '$scope', '$http', '$rootScope', '$window',
+  'changeListService', 'editabilityService',
+  'explorationRightsService', 'explorationWarningsService',
+  'explorationSaveService',
+  function(
+      $scope, $http, $rootScope, $window,
+      changeListService, editabilityService,
+      explorationRightsService, explorationWarningsService,
+      explorationSaveService) {
+    $scope.isSaveInProgress = false;
+
+    $scope.isPrivate = function() {
+      return explorationRightsService.isPrivate();
+    };
+
+    $scope.isExplorationLockedForEditing = function() {
+      return changeListService.isExplorationLockedForEditing();
+    };
+
+    $scope.isEditableOutsideTutorialMode = function() {
+      return editabilityService.isEditableOutsideTutorialMode();
+    };
+
+    $scope.countWarnings = function() {
+      return explorationWarningsService.countWarnings();
+    };
+
+    $scope.discardChanges = function() {
+      explorationSaveService.discardChanges();
+    };
+
+    $scope.getChangeListLength = function() {
+      return changeListService.getChangeList().length;
+    };
+
+    $scope.isExplorationSaveable = function() {
+      return explorationSaveService.isExplorationSaveable();
+    };
+
+    $scope.isPublishModalOpening = function() {
```

Comment:
Ah, I see the issue, you are trying to track just the second modal.

Actually, I think we had this "is opening" check before the first (metadata) modal was introduced. So let's think about what it's meant to achieve. From the HTML I think it's just meant to show dots while stuff is in progress? If that's the case then maybe we should just show dots until both modals close and control returns to the controller? Plus, this feels like a very "UI thing", so it shouldn't be part of the logic-layer service.

There might be other cases where you might need to examine the use case to see whether what's going on still makes sense.